src/client.c

```diff
@@ -65,26 +81,43 @@ urlinfo_t *parse_url(char *url)
 int send_request(int fd, char *hostname, char *port, char *path)
 {
   const int max_request_size = 16384;
+  char send_buff[1025];
   char request[max_request_size];
   int rv;
+  // printf("send hostname")
+  sprintf(request,  "GET /%s HTTP/1.1\nHost: %s:%s\r\nConnection: close\r\n\r\n", path, hostname, port);
+  
+  // printf("%s \n", request);
 
-  ///////////////////
-  // IMPLEMENT ME! //
-  ///////////////////
+  send(fd, request, strlen(request), 0);
 
   return 0;
 }
 
 int main(int argc, char *argv[])
 {  
   int sockfd, numbytes;  
-  char buf[BUFSIZE];
+  char buf[2056];
 
   if (argc != 2) {
     fprintf(stderr,"usage: client HOSTNAME:PORT/PATH\n");
     exit(1);
   }
+  // parse_url(argv[1]);
+  urlinfo_t *urlinfo = parse_url(argv[1]);
+  // printf("int main path: %s\n", urlinfo->path);
+
+  // printf("int main hostname: %s \n", urlinfo_t.hostname->hostname);
+  sockfd = get_socket(urlinfo->hostname, urlinfo->port);
+
+  send_request(sockfd, urlinfo->hostname, urlinfo->port, urlinfo->path);
 
+    numbytes = recv(sockfd,buf,sizeof(buf)-1,0); // <-- -1 to leave room for a null terminator
+    buf[numbytes] = 0; // <-- add the null terminator
+    printf("recv()'d %d bytes of data in buf\n",numbytes);
+    printf("%s",buf);
+  // while ((numbytes = recv(sockfd, buf, sizeof(buf) - 1, 0)) > 0) {
```

Comment:
You had the right idea here. Just move your `printf("%s",buf);` inside the while loop.

Also, make sure you're freeing up allocated memory and closing sockets.